kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/AdminClientProvider.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.admin.kafka.admin;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.kafka.admin.Config;
+import io.vertx.kafka.admin.KafkaAdminClient;
+import io.vertx.kafka.admin.ListOffsetsResultInfo;
+import io.vertx.kafka.admin.NewTopic;
+import io.vertx.kafka.admin.OffsetSpec;
+import io.vertx.kafka.admin.TopicDescription;
+import io.vertx.kafka.client.common.ConfigResource;
+import io.vertx.kafka.client.common.TopicPartition;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * class for admin client provider
+ */
+public class AdminClientProvider {
```

Comment:
I am a bit confused about the role or the naming of this class. In the end a "provider" should provide an instance of the admin client while this is a wrapper where all methods just call the admin client methods.